### PR TITLE
Strawberry perl windows sysread function returns POSIX error codes

### PR DESCRIPTION
**Pull request by ralph-irving**
whereas ActiveState perl returns windows sockets (WSA) error codes, support both.

### CODE REVIEW

**Review by mherger — COMMENTED** [review]
Would it make sense to have a flag like `main::ISACTIVEPERL` to make these conditions more flexible and readable? Otherwise we take a decision based on a Perl version - which might change in the future.

See eg. https://stackoverflow.com/questions/37056509/how-to-detect-activestate-version-of-perl

**Comment by ralph-irving** [comment]
Yes, I think that would be a better approach.  I'll take a look at implementing it.

File: Slim/Utils/Errno.pm [hunk 1]
```diff
@@ -39,11 +39,19 @@ our @EXPORT = qw(EWOULDBLOCK EINPROGRESS EINTR ECHILD EBADF);
...
+			*EBADF       = sub () { 10009 };
+			*ECHILD      = sub () { 10010 };
+			*EWOULDBLOCK = sub () { 10035 };
+			*EINPROGRESS = sub () { 10036 };
+		} else {
+			*EINTR       = sub () { 4 };
+			*EBADF       = sub () { 9 };
+			*ECHILD      = sub () { 10 };
+			*EWOULDBLOCK = sub () { 140 };
+			*EINPROGRESS = sub () { 112 };
```
Comment (michaelherger) [inline Slim/Utils/Errno.pm:53]: Are these really different from the POSIX values? Or is this a mistake?

According to https://perldoc.perl.org/Errno.txt:

```
EWOULDBLOCK => 11,
EINPROGRESS => 115,
```

Comment (michaelherger) [reply Slim/Utils/Errno.pm:53]: Ok, that's probably a difficult topic. Found eg. https://www.nntp.perl.org/group/perl.perl5.porters/2015/03/msg226651.html

Never mind.

Comment (ralph-irving) [reply Slim/Utils/Errno.pm:53]: If those defines were wrong for strawberry perl, then LMS wouldn't work as both are referenced in the code base.  I spent a lot of time researching this when I first tried to get LMS working with SBP.   I had to add a block of code to display the values associated with those defines when I was troubleshooting why LMS locked up when started with SBP.  So at least for SBP v5.32 they are correct.

**Pull request merged**